metadata-ingestion/src/datahub/ingestion/source/kafka_connect.py

```diff
@@ -879,6 +929,153 @@ def _extract_lineages(self):
+        if "snowflake.topic2table.map" in connector_manifest.config:
+            for each in connector_manifest.config["snowflake.topic2table.map"].split(
+                ","
+            ):
+                topic, table = each.split(":")
+                # Extract lineage for only those topics whose data ingestion started
+                if topic in connector_manifest.topic_names:
+                    topics_to_tables[topic] = table
+        else:
+            for topic in connector_manifest.topic_names:
+                topics_to_tables[topic] = topic
+
+        return self.SnowflakeParser(
+            database_name=database_name,
+            schema_name=schema_name,
+            topics_to_tables=topics_to_tables,
+        )
+
+    def _extract_lineages(self):
+        self.connector_manifest.flow_property_bag = self.connector_manifest.config
+        # remove private key from properties
+        del self.connector_manifest.flow_property_bag["snowflake.private.key"]
+
+        lineages: List[KafkaConnectLineage] = list()
+        parser = self.get_parser(self.connector_manifest)
+
+        for topic, table in parser.topics_to_tables.items():
+            target_dataset = f"{parser.database_name}.{parser.schema_name}.{table}"
+            lineages.append(
+                KafkaConnectLineage(
+                    source_dataset=topic,
+                    source_platform=KAFKA,
+                    target_dataset=target_dataset,
+                    target_platform="snowflake",
+                )
+            )
+
+        self.connector_manifest.lineages = lineages
+        return
+
+    def _extract_cll(self):
+        for lineage in self.connector_manifest.lineages:
+            # If sink type is file or
+            # Source dataset is none
+            if self.ctx.graph is None or lineage.source_dataset is None:
+                continue
+
+            source_dataset_run = make_lineage_dataset_urn(
+                self.config,
+                lineage.source_platform,
+                lineage.source_dataset,
+                get_platform_instance(
+                    self.config, self.connector_manifest.name, lineage.source_platform
+                ),
+            )
+
+            source_schema_aspect = self.ctx.graph.get_aspect(
+                source_dataset_run,
+                SchemaMetadataClass,
+            )
+            if source_schema_aspect is None:
+                # If source dataset metadata is not yet ingested
+                continue
+
+            is_snowpipe_streaming: bool = False
+            if "snowflake.ingestion.method" in self.connector_manifest.config:
+                is_snowpipe_streaming = self.connector_manifest.config[
+                    "snowflake.ingestion.method"
+                ]
+
+            # key schema dosn't get mapped with any target dataset column
+            source_dataset_fields: List[str] = [
+                schema_field.fieldPath.split(".")[-1]
```

Comment:
What's this split and take the last entry for ? This doesn't seem right.

Wonder how structs or nested structs or unions are mapped in snowflake table.